### ISSUE TITLE
no_std support?

### ISSUE THREAD

**Issue by berkus** [LukasKalbertodt/libtest-mimic#21]
I am using a custom test harness for OSdev, but wondering if this could do no_std as well?

**Comment by LukasKalbertodt** [comment]
Hi! I don't quite understand the use case yet. Is it a requirement to run the tests in a no-std environment? Usually for tests you can just use your development machine, no? Care to elaborate on your use case?

Apart from that, I don't really see how this library could become no_std. For one, it parses command line arguments via `clap`: `clap` doesn't yet support no_std and how would that work without `std::env` anyway? Also, libtest-mimic uses multi-threading to run tests in parallel. That couuuld in theory be gated behind a crate feature, but I don't think I woudl want the extra maintenance overhead.

**Comment by berkus** [comment]
Yes, the tests run in the no_std environment, usually via QEMU or other emulator.

**Comment by berkus** [comment]
[This](https://os.phil-opp.com/testing/) is how it typically works.

**Comment by LukasKalbertodt** [comment]
Coming back to this issue, it now makes total sense to me to run tests in no_std environments. But again, what about the command line parsing and multi threading? I assume you are fine with not having that? And for example creating and configuring an `Arguments` value in code?

**Comment by berkus** [comment]
> Coming back to this issue, it now makes total sense to me to run tests in no_std environments. But again, what about the command line parsing and multi threading? I assume you are fine with not having that? And for example creating and configuring an `Arguments` value in code?

Correct, the no_std harness would just set up a configuration manually (or semiautomagically) without command line parsing. There are no_std arg parsers like [argtokens](https://lib.rs/crates/argtokens) but they are probably not even necessary.
